### ISSUE TITLE
Cannot read property IPAddress of undefined fwdRules.forEach

### ISSUE THREAD

**Issue by JeffGiroux** [F5Networks/f5-cloud-failover-extension#27]
Using terraform to deploy my environment. I have target instance groups created, a forwarding rule, but I do NOT use the forwarding rule address as a virtual server. For unrelated reasons, I'm simply using 0.0.0.0/0 as a test VS. But...the terraform resources for target instance and forwarding rules are still created.

When this happens...AND the BIG-IP is NOT using a virtual address from a forwarding rule, you get an error.
```
Sun, 14 Jun 2020 17:27:22 GMT - info: [f5-cloud-failover] Performing failover - execute
Sun, 14 Jun 2020 17:27:22 GMT - info: [f5-cloud-failover] Performing Failover - discovery
Sun, 14 Jun 2020 17:27:23 GMT - severe: [f5-cloud-failover] Cannot read property 'IPAddress' of undefined TypeError: Cannot read property 'IPAddress' of undefined
    at fwdRules.forEach (/var/config/rest/iapps/f5-cloud-failover/nodejs/providers/gcp/cloud.js:809:56)
    at Array.forEach (<anonymous>)
    at Promise.all.then (/var/config/rest/iapps/f5-cloud-failover/nodejs/providers/gcp/cloud.js:808:31)
    at tryCatcher (/usr/share/rest/node/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:638:18)
    at PromiseArray._resolve (/usr/share/rest/node/node_modules/bluebird/js/release/promise_array.js:126:19)
    at PromiseArray._promiseFulfilled (/usr/share/rest/node/node_modules/bluebird/js/release/promise_array.js:144:14)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:574:26)
    at Promise._settlePromise0 (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

This is related to GCP and forwarding rules in general...see my other issue https://github.com/F5Networks/f5-cloud-failover-extension/issues/26.

There needs to be better handling of the error and also possibly a "pass through" feature where...if error, still failover the remaining objects. I say this because...my alias IP and my routes did NOT failover due to the forwarding rule error. Once i fixed my VS to use an address from the forwarding rule, no error.

Suggestions:
1. Look for VS address that matches forwarding rule
2. If found, then search for target instances
3. Then update forwarding rule with other target instance from unit2

Behavior now...
1. Seems that the script will query for vm instances and find target instance groups
2. Then, script seems to expect remaining stuff of a target instance to be there too (forwarding rule, matching VS IP)
3. If step 2 fails to find a matching rule for a VM that lives in a target instance...fail
4. No other cloud objects move

## Workaround

Workaround for user...
1. If not planning to use a forwarding rule, then delete the target instance groups and the forwarding rule
2. Or...if not planning to use the IP address of the forwarding rule as your BIG-IP virtual server address, then delete the target instance groups and the forwarding rule

Workaround for CFE...
1. fail through...meaning, if forwarding rules fail is it possible to still move the other objects (aliasIP, routes)?
2. Update logic to not fail on forwarding rules if VS IP doesn't match. If latter, then that means user must have accidentally created a target instance in the past.

**Comment by shyawnkarim** [comment]
Thanks for suggesting this enhancement. We are now tracking this internally with issue ID AUTOSDK-363.

**Comment by jbaribeault** [comment]
I'm getting the same behaviour from the failover; BUT we have no forwarding rules at all; we just want the AliasIP to move. 

`Fri, 04 Sep 2020 13:25:38 GMT - finest: [f5-cloud-failover] Download stateFile: {"taskState":"SUCCEEDED","message":"Failover state file was reset","timestamp":"2020-09-04T13:24:50.946Z","instance":"none","failoverOperations":{}}
Fri, 04 Sep 2020 13:25:38 GMT - finest: [f5-cloud-failover] taskState: {"taskState":"SUCCEEDED","message":"Failover state file was reset","timestamp":"2020-09-04T13:24:50.946Z","instance":"none","failoverOperations":{}}
Fri, 04 Sep 2020 13:25:38 GMT - fine: [f5-cloud-failover] Address operations enabled?  true
Fri, 04 Sep 2020 13:25:38 GMT - fine: [f5-cloud-failover] Route operations enabled?  false
Fri, 04 Sep 2020 13:25:38 GMT - info: [f5-cloud-failover] Performing failover - execute
Fri, 04 Sep 2020 13:25:40 GMT - finest: [f5-cloud-failover] State file data:  {"taskState":"SUCCEEDED","message":"Failover state file was reset","timestamp":"2020-09-04T13:24:50.946Z","instance":"none","failoverOperations":{}}
Fri, 04 Sep 2020 13:25:40 GMT - finest: [f5-cloud-failover] Data will be uploaded to f5cloudfailoverstate.json:  {"taskState":"RUNNING","message":"Failover running","timestamp":"2020-09-04T13:25:40.227Z","instance":"bigip1-gccsplb00.c.pj-prd-ntw-nane1-01.internal","failoverOperations":{}}
Fri, 04 Sep 2020 13:25:40 GMT - info: [f5-cloud-failover] Performing Failover - discovery
Fri, 04 Sep 2020 13:25:40 GMT - fine: [f5-cloud-failover] Getting failover addresses using selfAddresses  {"0":{"address":"10.4.64.46","trafficGroup":"/Common/traffic-group-local-only","trafficGroupMatch":false}}  and floatingAddresses  {"0":{"address":"10.4.64.51"}}
Fri, 04 Sep 2020 13:25:40 GMT - fine: [f5-cloud-failover] Retrieved local addresses {"0":"10.4.64.46"}
Fri, 04 Sep 2020 13:25:40 GMT - fine: [f5-cloud-failover] Retrieved failover addresses  {"0":"10.4.64.51"}
Fri, 04 Sep 2020 13:25:40 GMT - finest: [f5-cloud-failover] updateAddresses:  {"localAddresses":["10.4.64.46"],"failoverAddresses":["10.4.64.51"],"discoverOnly":true}
Fri, 04 Sep 2020 13:25:41 GMT - fine: [f5-cloud-failover] Failover addresses to discover {"0":"10.4.64.51"}
Fri, 04 Sep 2020 13:25:41 GMT - finest: [f5-cloud-failover] VM name: bigip2-gccsplb00
Fri, 04 Sep 2020 13:25:41 GMT - finest: [f5-cloud-failover] getFwdRules called with nextPageToken
Fri, 04 Sep 2020 13:25:41 GMT - severe: [f5-cloud-failover] Cannot read property 'IPAddress' of undefined TypeError: Cannot read property 'IPAddress' of undefined
    at fwdRules.forEach (/var/config/rest/iapps/f5-cloud-failover/nodejs/providers/gcp/cloud.js:809:56)
    at Array.forEach (<anonymous>)
    at Promise.all.then (/var/config/rest/iapps/f5-cloud-failover/nodejs/providers/gcp/cloud.js:808:31)
    at tryCatcher (/usr/share/rest/node/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:638:18)
    at PromiseArray._resolve (/usr/share/rest/node/node_modules/bluebird/js/release/promise_array.js:126:19)
    at PromiseArray._promiseFulfilled (/usr/share/rest/node/node_modules/bluebird/js/release/promise_array.js:144:14)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:574:26)
    at Promise._settlePromise0 (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
Fri, 04 Sep 2020 13:25:41 GMT - finest: [f5-cloud-failover] Data will be uploaded to f5cloudfailoverstate.json:  {"taskState":"FAILED","message":"Failover failed because Cannot read property 'IPAddress' of undefined","timestamp":"2020-09-04T13:25:41.326Z","instance":"bigip1-gccsplb00.c.pj-prd-ntw-nane1-01.internal","failoverOperations":{"addresses":null,"routes":null}}
Fri, 04 Sep 2020 13:25:46 GMT - finest: socket 243 closed
`

**Comment by shyawnkarim** [comment]
Closing.

The enhancement was included with [Release 1.7](https://github.com/F5Networks/f5-cloud-failover-extension/releases/tag/v1.7.0).
